example/integration_test/sign_tx_test_cases.dart

```diff
@@ -0,0 +1,496 @@
+import 'package:ledger_cardano/ledger_cardano.dart';
+import 'package:ledger_cardano/src/models/script_data_hash.dart';
+import 'package:ledger_cardano/src/models/transaction_signing_mode.dart';
+import 'package:ledger_cardano/src/models/witness.dart';
+import 'package:ledger_cardano/src/models/parsed_address_params.dart';
+import 'package:ledger_cardano/src/models/shelley_address_params.dart';
+import 'package:ledger_cardano/src/models/spending_data_source.dart';
+import 'package:ledger_cardano/src/models/staking_data_source.dart';
+import 'package:ledger_cardano/src/models/parsed_datum.dart';
+import 'package:ledger_cardano/src/models/parsed_token.dart';
+import 'package:ledger_cardano/src/models/parsed_required_signer.dart';
+import 'package:ledger_cardano/src/models/parsed_certificate.dart';
+import 'package:ledger_cardano/src/models/parsed_credential.dart';
+import 'package:ledger_cardano/src/models/parsed_withdrawal.dart';
+
+import 'test_utils.dart';
+
+class SignTxTestCase {
+  final String testName;
+  final ParsedSigningRequest request;
+  final SignedTransactionData expected;
+
+  SignTxTestCase({
+    required this.testName,
+    required this.request,
+    required this.expected,
+  });
+}
+
+final testsAlonzo = [
+  SignTxTestCase(
+    testName: 'Sign tx with script data hash',
+    request: ParsedSigningRequest(
+      tx: ParsedTransaction(
+        network: CardanoNetwork.mainnet(),
+        fee: BigInt.from(42),
+        ttl: BigInt.from(10),
+        inputs: [
+          ParsedInput(
+            txHashHex: '3b40265111d8bb3c3c608d95b3a0bf83461ace32d79336579a1939b3aad1c0b7',
+            outputIndex: 0,
+            path: [harden + 1852, harden + 1815, harden + 0, 0, 0],
```

Comment:
I don't particularly like that we send here the full path. Feels very error-prone. We should probably improve here the API to some freezed class

sealed class LedgerSigningPath, with options for Byron, ShelleyPayment, ShelleyStake, Custom.

For ParsedTransaction, `additionalWitnessPaths` should use this same `LedgerSigningPath`